Add expanded_ti_count to ti context

uranusjr opened the pull request:
A very small part of #27678 that can go in independently.

This value indicates how many tis the current mapped task is mapped into. This will be used to resolve values from parent a mapped task group without needing to call the retrival function repeatedly (which incurs database overhead).

uranusjr committed:
Add expanded_ti_count to ti context

This value indicates how many tis the current mapped task is mapped
into. This will be used to resolve values from parent a mapped task
group without needing to call the retrival function repeatedly (which
incurs database overhead).

potiuk commented on docs/apache-airflow/templates-ref.rst:
```diff
@@ -75,6 +75,8 @@ Variable                                    Description
 ``{{ dag_run }}``                           A reference to the DagRun object.
 ``{{ test_mode }}``                         Whether the task instance was called using the CLI's test
                                             subcommand.
+``{{ expanded_ti_count }}``                 Number of task instances that a mapped task was expanded into. If
```
I guess we should we add "new in 2.5" somewhere?

uranusjr replied:
I thought about that but it doesn’t seem to fit very well with the current format, and many other variables here (e.g. the data interval ones) don’t annotate when they were added either. Maybe I can revamp the entire table?

uranusjr replied:
I restructured the entire table.

uranusjr committed:
Remake template variable table

Type information is separated into a separate column, and some
additional information is added to the description.
